icu4c/source/i18n/ucol_res.cpp

```diff
@@ -392,6 +392,7 @@ CollationLoader::loadFromData(UErrorCode &errorCode) {
     // but that created undesirable dependencies.
     int32_t length;
     const uint8_t *inBytes = ures_getBinary(binary.getAlias(), &length, &errorCode);
+    if(U_FAILURE(errorCode)) { return nullptr; }
```

Comment:
optional -- simpler:
```suggestion
    int32_t length = 0;
    const uint8_t *inBytes = ures_getBinary(binary.getAlias(), &length, &errorCode);
```